logging/interceptor.go

```diff
@@ -181,9 +237,35 @@ func addCustomField(ctx context.Context, fields logrus.Fields, customField strin
 		return ctx, fmt.Errorf("Unable to get custom %q field from context", customField)
 	}
 
+	// In case of subject field is a map
+	if customField == DefaultSubjectKey {
+
+		replacer := strings.NewReplacer("map[", "", "]", "")
+		field = replacer.Replace(field)
+		inner := strings.Split(field, " ")
+
+		m := map[string]interface{}{}
+
+		for _, v := range inner {
+			kv := strings.Split(v, ":")
+
+			if len(kv) == 1 {
+				fields[customField] = kv[0]
+
+				return ctx, err
+			}
+
+			m[kv[0]] = kv[1]
+		}
+
+		fields[customField] = m
+
+		return ctx, err
+	}
+
 	fields[customField] = field
 
-	return metadata.AppendToOutgoingContext(ctx, customField, field), err
+	return ctx, err
```

Comment:
If we are not changing anything in context, can we remove it from return statement ?